control-plane-operator/controllers/hostedcontrolplane/etcd/reconcile.go

```diff
@@ -89,6 +101,38 @@ func ReconcileStatefulSet(ss *appsv1.StatefulSet, p *EtcdParams) error {
 	return nil
 }
 
+func buildEtcdInitContainer(p *EtcdParams) func(c *corev1.Container) {
+	return func(c *corev1.Container) {
+		// FIXME: etcdctl restore is deprecated but the etcd container doesn't have etcdutl
+		script := `
+[ "$(ls -A /var/lib/data)" ] && echo "/var/lib/data not empty, not restoring snapshot" && exit 0
```

Comment:
@csrwng and @derekwaynecarr asked some questions about the mutability of `restoreSnapshotURL` and what happens if the etcd pod gets restarted - my intention here is that the URL is immutable, and only gets consumed on the very first pod startup.

After the initial startup, `/var/lib/data` will not be empty, so we'll exit early here and not restore anything leaving the existing data intact

Reply:
Can we may be ensure the expectations and workflow you describe above is also enforced programatically by comparing spec with status? e.g if .restoreSnapshotURL succeeded (criteria: /usr/bin/etcdctl snapshot restore succeed and etc pod is healthy?) then status.conditions[etcdRestored] = true

Reply:
@enxebre that's a nice idea, thanks - yes I'll update this to expose the status, which will be clearer on pod restart than having an initContainer which does nothing :+1: